browser/caliban.py

```diff
@@ -621,44 +594,27 @@ def readable_tracks(self):
 
         return tracks
 
-    @property
-    def png_colormap(self):
-
 
-        return {str(self.color_map(v, bytes=True)): int(v)
-                  for v in range(max(self.tracks) + 1)}
-
-    def get_frame(self, frame, raw, edit_background):
+    def get_frame(self, frame, raw):
         self.current_frame = frame
         if raw:
-            frame = self.trial["raw"][frame][:,:,0]
+            frame = self.raw[frame][:,:,0]
             return pngify(imgarr=frame,
                           vmin=0,
                           vmax=None,
                           cmap="cubehelix")
-        if edit_background:
-            frame = self.trial["raw"][frame][:,:,0]
-            return pngify(imgarr=frame,
-                          vmin=0,
-                          vmax=None,
-                          cmap="Greys")
-
         else:
-
-            frame = self.trial["tracked"][frame][:,:,0]
-            self.color_map.set_bad('red')
-
-            if (self.highlight):
-                if (self.highlight_cell_one != -1):
-                    frame = np.ma.masked_equal(frame, self.highlight_cell_one)
-                if (self.highlight_cell_two != -1):
-                    frame = np.ma.masked_equal(frame, self.highlight_cell_two)
-
+            frame = self.tracked[frame][:,:,0]
+            frame = np.ma.masked_equal(frame, 0)
             return pngify(imgarr=frame,
                          vmin=0,
-                         vmax=self.num_tracks,
+                         vmax=max(self.tracks),
```

Comment:
this could be problematic if a large label is chosen. Ideally i think we get the length of unique ids.

Reply:
Moved investigation to issue  #73